### PR TITLE
Fixes/pp 71/fix allow tokens scripts

### PR DESCRIPTION
**Pull request by wilsoniovlabs**
It seems there are some issues with the current version of the allowTokens and allowedTokens scripts, under the rif-relay-contracts repo.

It appears that:

network names are used during deployment (e.g. development) but IDs are used both in the contract-addresses.json file as well as the tokens scripts.

the allowedTokens script uses $2 as the network parameter name, instead of $1.

not all errors are displaying in the console in case something goes wrong.

Verify and fix this.

Also: update the Sample dApp readme so that the allow tokens section points to the rif-relay-contracts repo, once it works OK.

### CODE REVIEW

**Review by mortelli — APPROVED** [review]
looks good, and seems to work OK :+1: 

there's 2 more things we can do here:
1. add the `NETWORK_NAME` parameter to line `121` of the readme file for this project (the part which explains how to run `allowedTokens`)
2. open a small PR in the sample dapp repo, changing its readme file as the description for PP-71 indicates

thanks

**Comment by mortelli** [comment]
@antomor please review when you get a chance :+1:

File: scripts/allowedTokens [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-NETWORK_ID=$2
+NETWORK_ID=$1
```
Comment (antomor) [inline scripts/allowedTokens:3]: I would rename this variable to `NETWORK_NAME` since that it's not anymore a network id

**Review by antomor — COMMENTED** [review]
LGTM!

**[OPTIONAL]**: I would change the name of the file generated: https://github.com/anarancio/rif-relay-contracts/pull/4/files#diff-599d0b98ec67caefca50885fa601dbf862f6ec8d762c93eff96df7aeea30046eR9

I know it was already there, but I would create a file named `allowedTokens` instead of `allowToken` or eventually I would prefer something like `temp.js`

**Review by antomor — APPROVED** [review]

File: scripts/allowedTokens (hunk 1 above)
Comment (mortelli) [reply scripts/allowedTokens:3]: agreed

**Review by mortelli — COMMENTED** [review]
@antomor
> I would change the name of the file generated: https://github.com/anarancio/rif-relay-contracts/pull/4/files#diff-599d0b98ec67caefca50885fa601dbf862f6ec8d762c93eff96df7aeea30046eR9
> 
> I know it was already there, but I would create a file named `allowedTokens` instead of `allowToken` or eventually I would prefer something like `temp.js`

sorry, which file do you mean? i am following the link but it's showing nothing in particular, just the diff it seems

**Comment by antomor** [comment]
> I would change the name of the file generated: https://github.com/anarancio/rif-relay-contracts/pull/4/files#diff-599d0b98ec67caefca50885fa601dbf862f6ec8d762c93eff96df7aeea30046eR9
 I know it was already there, but I would create a file named allowedTokens instead of allowToken or eventually I would prefer something like temp.js


Sorry, I can see a yellow line... However I was referencing line #9 of `scripts/allowedTokens`:
```
cat > allowTokens.js << EOF
```

File: scripts/allowedTokens [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-NETWORK_ID=$2
+NETWORK_NAME=$1
 
-if [ "${NETWORK_ID}" == "" ]; then
-        NETWORK_ID="regtest"
+if [ "${NETWORK_NAME}" == "" ]; then
+        NETWORK_NAME="regtest"
 fi
 
 cat > allowTokens.js << EOF
```
Comment (antomor) [inline scripts/allowedTokens:9]:
```suggestion
cat > allowedTokens.js << EOF
```

Comment (antomor) [inline scripts/allowedTokens:9]:
```suggestion
cat > allowedTokens.js << EOF
```

File: scripts/allowedTokens [hunk 3]
```diff
@@ -56,6 +56,6 @@ module.exports = async function(callback) {
 }
 EOF
 
-truffle exec --network ${NETWORK_ID} allowTokens.js
+truffle exec --network ${NETWORK_NAME} allowTokens.js
 
 rm allowTokens.js
```
Comment (antomor) [inline scripts/allowedTokens:61]:
```suggestion
rm allowedTokens.js
```

File: scripts/allowedTokens (hunk 2 above)
Comment (mortelli) [reply scripts/allowedTokens:9]: agreed

File: scripts/allowedTokens [hunk 4]
```diff
@@ -56,6 +56,6 @@ module.exports = async function(callback) {
 }
 EOF
 
-truffle exec --network ${NETWORK_ID} allowTokens.js
+truffle exec --network ${NETWORK_NAME} allowTokens.js
```
[the remark this answers is not in the document]
Comment (mortelli) [reply scripts/allowedTokens:59]: same here

File: scripts/allowedTokens (hunk 3 above)
Comment (mortelli) [reply scripts/allowedTokens:61]: and here

File: contract-addresses.json [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
+        "smartWallet": "0x0bf88c200cc8f5f2cA409a35adf69dcbFc2e6Caf",
+        "smartWalletFactory": "0x2Ec01561f6F2e2325630A58Eb1Be43c51d7bdf70",
+        "smartWalletDeployVerifier": "0xA5344b32BFeEB51bf70693bBFECe3F23d82c39bE",
+        "smartWalletRelayVerifier": "0x69788F628C5Db69AC37506573162Ac15C735f726",
+        "customSmartWallet": "0xAED72be45dd3206635183013E910f415c9732CCa",
+        "customSmartWalletFactory": "0xE9797325c5C3A841B4C4f03D15334e5401f9F78F",
+        "customSmartWalletDeployVerifier": "0x545c8EEA346855eF6961346dCc1568a1a028EA50",
+        "customSmartWalletRelayVerifier": "0xf02976B499c43FfC1Ec16611581e96C2A750ADcf",
+        "sampleRecipient": "0xb4ed846447053EA8502E9292edbC8620CAe3F386",
+        "testToken": "0x896A78a1d3E37588079D984291b09532916ae6CB"
```
[the remark this answers is not in the document]
Comment (wilsoniovlabs) [reply contract-addresses.json]: This was fixes

File: scripts/allowedTokens (hunk 1 above)
Comment (wilsoniovlabs) [reply scripts/allowedTokens]: Fixed.

File: scripts/allowTokens [hunk 6]
```diff
@@ -86,6 +86,7 @@ module.exports = async function(callback) {
       }
     }
   } catch (error) {
+    console.error(error);
     console.error("Fail to allow tokens");
```
[the remark this answers is not in the document]
Comment (wilsoniovlabs) [reply scripts/allowTokens]: Fixed.

File: contract-addresses.json [hunk 7]
```diff
@@ -1,16 +1 @@
...
...ED72be45dd3206635183013E910f415c9732CCa","customSmartWalletFactory":"0xE9797325c5C3A841B4C4f03D15334e5401f9F78F","customSmartWalletDeployVerifier":"0x545c8EEA346855eF6961346dCc1568a1a028EA50","customSmartWalletRelayVerifier":"0xf02976B499c43FfC1Ec16611581e96C2A750ADcf","sampleRecipient":"0xb4ed846447053EA8502E9292edbC8620CAe3F386","testToken":"0x896A78a1d3E37588079D984291b09532916ae6CB"},"regtest":{"penalizer":"0xe8C7C6f18c3B9532343487faD807060750C1fE95","relayHub":"0x3bA95e1cccd397b5124BcdCC5bf0952114E6A701","smartWallet":"0xB7a001eE69E7C1eef25Eb8e628e46214Ea74BF0F","smartWalletFactory":"0x8C1108cFCd7ddad09D8910e5f42982A6c54aD9cD","smartWalletDeployVerifier":"0x1938517B0762103d52590Ca21d459968c25c9E67","smartWalletRelayVerifier":"0x74Dc4471FA8C8fBE09c7a0C400a0852b0A9d04b2","customSmartWallet":"0x96A90Ee24C20c78Ba20AcE1a2aa4D59F79353C54","customSmartWalletFactory":"0x89bac3BB0517F7Dc0E5E94265217A9Acc5cc489f","customSmartWalletDeployVerifier":"0x3eE31F6049065B616f85470985c0eF067f2bEbDE","customSmartWalletRelayVerifier":"0xDE8Ae20488BE104f0782C0126038b6682ECc1eC7","sampleRecipient":"0xB08E0D333F139AC71DE04f4bfE7e75ca1e6122a0","testToken":"0x5D0aBdE7Ed6B7e122eD55EAe514E617d6c08f407"}}
```
[the remark this answers is not in the document]
Comment (wilsoniovlabs) [reply contract-addresses.json]: Fixed.

File: scripts/allowedTokens (hunk 2 above)
Comment (wilsoniovlabs) [reply scripts/allowedTokens:9]: Fixed.

File: scripts/allowedTokens (hunk 3 above)
Comment (wilsoniovlabs) [reply scripts/allowedTokens:61]: Fixed.

File: scripts/allowedTokens (hunk 4 above)
[the remark this answers is not in the document]
Comment (wilsoniovlabs) [reply scripts/allowedTokens]: Fixed.

**Comment by antomor** [comment]
LGTM!

**Comment by mortelli** [comment]
@wilsoniovlabs 

go ahead and merge :+1: 

@antomor 

i've created 2 follow-up PRs for these things:
> there's 2 more things we can do here:
> 1. add the `NETWORK_NAME` parameter to line `121` of the readme file for this project (the part which explains how to run `allowedTokens`)
> 2. open a small PR in the sample dapp repo, changing its readme file as the description for PP-71 indicates

these are:
- https://github.com/anarancio/rif-relay-contracts/pull/6
- https://github.com/rsksmart/rif-relay-sample-dapp/pull/15
 
please review when you get a chance (after this PR is merged)

**Comment by mortelli** [comment]
@wilsoniovlabs 

regarding commit [8706928](https://github.com/anarancio/rif-relay-contracts/commit/87069282bf1a319f54a62e2a6be5c77566e5cc44):

there's 2 types of errors:
1. general script errors
2. tx revert errors, which are a subset of the first kind. these will include a receipt field, and in these cases we'd be using the `getRevertReason` function

this is why i think we should keep this function around. what we can do is add code to check if the caught error has the `receipt` property, and if so, call the `getRevertReason` method. if not, then display the error as it comes

**Pull request merged**